### PR TITLE
Lab1

### PR DESCRIPTION
**Pull request by ezio-42**

### CODE REVIEW

**Comment by ezio-42** [comment]
@bkmceh @danmory please review

File: app_haskell/HASKELL.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+* Use version control system (git)
+* Use dependency management tool (stack)
+* Follow style guidelines (hlint)
+* Use libraries from Hackage
+* Use appropriate .gitignore file
+* Use pre-commit
+  * hlint           - linter
+  * stylish-haskell - auto code formatter
+* Use production ready libraries
+  * servant - most popular library for building REST APIs.
```
Comment (danmory) [inline app_haskell/HASKELL.md:13]: Is popularity the only reason?

File: app_haskell/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# app_haskell
```
Comment (danmory) [inline app_haskell/README.md]: The name of the project tells nothing about it

File: app_haskell/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## CI
+
+### stylish-haskell
+
+You can quickly download the latest binary and run `stylish-haskell` like so:
+
+    curl -sL https://raw.github.com/haskell/stylish-haskell/master/scripts/latest.sh | sh -s .
+
+Where the `.` can be replaced with the arguments you pass to stylish-haskell.
```
Comment (danmory) [inline app_haskell/README.md]: If `.` is used to indicate place where to paste specific parameters then better to use something like sh -s [your_argument]

File: app_python/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# app_python
```
Comment (danmory) [inline app_python/README.md]: The name of the project tells nothing about it

File: app_python/PYTHON.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+* Use data classes
+* Use appropriate .gitignore file
+* Use type checker (mypy)
+* Use pre-commit
+  * black  - auto code formatter and linter
+  * flake8 - linter and error checker
+  * isort  - sort imports
+  * mypy   - type checker
+* Use production ready libraries
+  * Aiohttp - is asynchronous library which provides server and client functionality. Used by large companies ([link](https://docs.aiohttp.org/en/stable/powered_by.html)).
```
Comment (danmory) [inline app_python/PYTHON.md]: Is `Used by large companies` the only reason why aiohttp is production ready?

File: app_python/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    black --check . --verbose --diff
+
+#### Automatically fix formatting (done in pre-commit)
+
+    black .
+
+### Run flake8
+
+    flake8 --verbose --max-line-length=80 --ignore="E203,W503"
```
Comment (danmory) [inline app_python/README.md:41]: Why to ignore these rules?

File: app_python/PYTHON.md (hunk 5 above)
Comment (bkmceh) [inline app_python/PYTHON.md]: I didnt understand why for this app your choice is Aiohttp, can you please extend this topic?

File: app_haskell/HASKELL.md (hunk 1 above)
Comment (bkmceh) [reply app_haskell/HASKELL.md:13]: LGTM excluding top comment

File: app_haskell/README.md (hunk 2 above)
Comment (ezio-42) [reply app_haskell/README.md]: fixed

File: app_python/README.md (hunk 4 above)
Comment (ezio-42) [reply app_python/README.md]: fixed

File: app_python/README.md (hunk 6 above)
Comment (ezio-42) [reply app_python/README.md:41]: due to conflict between `black` and `flake8`

File: app_python/PYTHON.md (hunk 5 above)
Comment (ezio-42) [reply app_python/PYTHON.md]: README fixed

File: app_haskell/README.md (hunk 3 above)
Comment (ezio-42) [reply app_haskell/README.md]: fixed

File: app_haskell/HASKELL.md (hunk 1 above)
Comment (ezio-42) [reply app_haskell/HASKELL.md:13]: fixed

**Comment by ezio-42** [comment]
all questions are resolved

File: app_python/PYTHON.md (hunk 5 above)
Comment (ezio-42) [reply app_python/PYTHON.md]: no, README fixed

**Pull request closed without merging**